docs/data/toolpad/getting-started/why-toolpad.md

```diff
@@ -0,0 +1,81 @@
+- **Less boilerplate** so you can focus on the essential parts of the app.
+- **Run alongside existing code**. It integrates well with your existing code. You can use your database models, client libraries, secrets, and bespoke components directly.
+- **Single source of truth**. Because Toolpad runs alongside your existing code, it lets you maintain a single source of truth for all business logic.
+- **Integrates well with your development lifecycle**. Your project lives on your file system. You can use your own IDE, version control system, deployment target, CLI tools.
+
+On reviewing multiple products and approaches, a few recurring areas of concern were noticed. Let's examine them in detail:
+
+## Trade-offs with pro-code tools
+
+### Compromised delivery speed
+
+Every organization aspires to achieve rapid delivery, but to accommodate potential use cases, developers are often compelled to opt for flexible pro-code solutions, which typically come at the expense of speed. These tools offer lower abstraction levels for more control, but handling and scaling them soon becomes an unavoidable maintenance overhead.
+
+### More lines of code
+
+Lots of boilerplate HTML, CSS, and JavaScript glue code is needed just to get backend data to the browser and hooked up to data displaying components.
+
+### Lower-level abstraction
+
+Building internal tools requires knowledge of front-end framework. While it may interest some, most internal tool developers find the lower-level APIs, HTML and CSS an extra burden to maintain.
+
+## Trade-offs with low-code tools
+
+There exist many low-code code tools specifically for internal tools. They are proven to be productive but come with some trade-offs, as mentioned below:
+
+### Codebase fragmentation and duplication
+
+Frequently, developers invest valuable time in hunting down particular versions of business logic and then maintaining them in multiple places. When working with external low-code/no-code tools, code/logic duplication can often arise as a consequence of design decisions. Not having a unified reference point within a codebase compromises its scalability as well as ease of maintenance and debugging.
+
+### Extensibility
+
+Low-code tools are hard to extend. If there's something the tool can't do out of the box, you can't bring your own code with you and integrate it seamlessly.
+
+### Loss of control
+
+When working with external tools, developers often forfeit their autonomy and are constrained by the choices offered by the tool. A black-box element intervention takes away the control over the entire process, encompassing integrations, deployment, and version control. This lack of control puts the underlying sensitive data to risk. The security of customer data is critical, and the seriousness of this matter can't be overstated.
```

Comment:
```suggestion
When developers use external tools, they frequently sacrifice their autonomy and become restricted by the options provided by the tool. The introduction of a black-box element removes control over the entire process, including integrations, deployment, and version control. This can lead to a significant problem when it comes to the long-term maintainability of internal tools, as it becomes challenging to adapt, extend, or customize them to meet evolving needs.
```